server/src/domain/system-config/dto/system-config.dto.ts

```diff
@@ -38,6 +39,10 @@ export class SystemConfigDto {
   @ValidateNested()
   @IsObject()
   thumbnail!: SystemConfigThumbnailDto;
+
+  @IsBoolean()
+  @Transform(toBoolean)
+  isConfigFile!: boolean;
```

Comment:
At the moment I think it's technically possible to write to this value via the API (which would persist it to the database, which would probably mean a bad time). We should prevent that if we can.

Reply:
I know, yeah... But tbh I couldn't think of a way to avoid it. You could have different object models for requests and responses, but I don't think this is good practice... I guess the same issue turns up when manually deleting the isConfigFile value before pasting it to the clipboard in the front end.

Reply:
So that the flag would the fetched separately?

Reply:
`updateConfig` uses `Object.values(SystemConfigKey)` to determine what gets stored in the database. Since `isConfigFile` isn't in `SystemConfigKey`, I don't think it would get persisted to the database either way.

Personally though, I would change the typing here so `SystemConfigDto` is unchanged and make a new type that contains `SystemConfigDto` and `isConfigFile` as fields. This clarifies what `isConfigFile` is and isn't, both to the compiler and to contributors.

Reply:
So there would be an extra SystemConfigResponse model? 
Having `isConfigFile` as a header flag is bad practice, right?

Reply:
We can use a feature flag instead of this extra response setting imo.

Reply:
So we wait for your PR to be merged (what's the current state?), then we use the feature here and merge this PR?

Reply:
It's just been merged, so that should be available now.